### DISCUSSION TITLE
Custom regex match for filter syntax

### DISCUSSION THREAD

**Discussion by Nekidev** [carltongibson/django-filter#1507] [Q&A] [answered]
Is there a way to do custom match patterns for filter query params? For example, instead of using `price__gt`, being able to do `filter[price.gt]`.

Something like this:
```python3
import re

match = r"^filter\[([a-zA-Z]+).?([a-zA-Z]+)?\]"
def parse_filter(match, filter):
    if not re.match(match, filter):
        raise Exception("Invalid filter")
    search = re.search(match, filter)
    return search.group(1), search.group(2)

query_param = input("Filter > ")
field, lookup_expr = parse_filter(match, query_param)
print(f"Result: " + (f"{field}__{lookup_expr}" if lookup_expr else field))
```

Here, the `parse_filter` func will return the field where the filter should be applied and the lookup expression (or `None` if does not exist). In this way, I would be able to change the syntax of the query param filters according to what I need.

Is there a way to do this? Or the default `field__lookup` is the only one allowed?

**Comment by carltongibson** [comment, marked as answer]
This isn't built-in, but see [LookupChoiceFilter](https://django-filter.readthedocs.io/en/stable/ref/filters.html#lookupchoicefilter) that does something similar. (You'll need to handle the mapping from the GET QueryDict to the `filter()` implementation.)
